ml_swissknife/db_utils.py

```diff
@@ -283,45 +258,254 @@ def append_df_to_db(
             raise e
 
 
-@contextmanager
-def create_connection(database, timeout=5.0, is_print=True, **kwargs):
-    """Create a database connection to a SQLite database"""
-    conn = None
-    try:
-        conn = sqlite3.connect(database, timeout=timeout, **kwargs)
-        if is_print:
-            logging.info(f"Connected to {database} SQLite")
-        yield conn
-    except sqlite3.Error:
-        logging.exception("Failed to connect with sqlite3 database:")
-    finally:
-        if conn:
-            conn.close()
+def get_primary_keys(database: Union[str, sa.engine.base.Engine], table_name: str) -> list[str]:
+    """Get the primary keys of a table in a database.
+
+    Parameters
+    ----------
+    database : str or engine
+        The URL to the database to connect to, or the sqlalchemy engine.
+
+    table_name : str
+        The name of the table to get the primary keys from.
+    """
+    engine = get_engine(database)
+    inspector = sa.inspect(engine)
+    return inspector.get_pk_constraint(table_name)["constrained_columns"]
+
+
+### Secondary helpers ###
+def get_table_info(database: Union[str, sa.engine.base.Engine], table_name: str) -> pd.DataFrame:
+    """Return a dataframe with the table information of table_name in a database."""
+    table = sa.Table(table_name, sa.MetaData(), autoload_with=db_utils.get_engine(database))
+
+    data = {
+        "name": [],
+        "type": [],
+        "primary_key": [],
+        "nullable": [],
+        "default": [],
+        "autoincrement": [],
+        "unique": [],
+    }
+
+    for column in table.columns:
+        data["name"].append(column.name)
+        data["type"].append(column.type)
+        data["primary_key"].append(column.primary_key)
+        data["nullable"].append(column.nullable)
+        data["default"].append(column.default)
+        data["autoincrement"].append(column.autoincrement)
+        data["unique"].append(column.unique)
+
+    return pd.DataFrame(data)
+
+
+def get_engine(database: Union[str, sa.engine.base.Engine], **engine_kwargs) -> sa.engine.base.Engine:
```

Comment:
nit: could we move this function to the top of the file for readability? this seems like the master function without which none of the others can work.